docs/design-docs/benchmarks.md

```diff
@@ -0,0 +1,274 @@
+---
+id: benchmarks
+---
+
+# Benchmarks Design
+
+## Goals
+
+The design of benchmarks has the following goals to achieve:
+
+**High Code Quality**
+* Extract the common code into the base class, reduce the workload to maintain all benchmarks.
+* Provide a unified entrance to launch benchmarks.
+
+**Good Extensibility**
+* Avoid modifying existing code when add new benchmarks by using registration mechanism.
+* Support benchmark-specific configuration, further support models with (same structure and different configs).
+
+**Good Usability**
+* Unify the output format for all the micro-benchmarks and E2E-benchmarks, contains return_code, metrics, raw-output, etc.
+* Can specify the run times for user to check the variance of one benchmark.
+
+## High-level Technical Design
+
+![Structure of Benchmark Package](../assets/benchmark-structure.png)
+*Figure 1 Structure of Benchmarks package*
+
+The structure of Benchmarks package can be divided into layers from the bottom up:
```

Comment:
Shall we add `` for all class name. e.g. `Benchmarks`

Reply:
OK